### ISSUE TITLE
More than 63 bosses will not add the rest

### ISSUE THREAD

**Issue by danylopez123** [50DKP/FF2-Official#470]
I was seeing what was wrong in my server that why some of my bosses didint loaded and then i realized, there was a limit of 63 bosses (Also counting the Duo ones)

This is my current list with the bosses in characters.cfg:

"Freak Fortress 2"
{
	"1"	"seeman"
	"2"	"seeldier"
	"3"	"demopan"
	"4"	"ninjaspy"
	"5"	"gentlespy"
	"6"	"saxtonhale"
	"7"	"vagineer_hale"
	"8"	"hhh_hale"
	"9"	"brutal_sniper_hale"
	"10"	"easterbunny"
	"11"	"mario2"
	"12"	"luigi2"
	"13"	"mario4_duo"
	"14"	"luigi4_duo"
	"15"	"Bowser"
	"16"	"sonic2"
	"17"	"tails2"
	"18"	"knuckles"
	"19"	"Shadow"
	"20"	"sonic_duo"
	"21"	"tails_duo"
	"22"	"wario"
	"23"	"waluigi"
	"24"	"slender"
	"25"	"Reimu"
	"26"	"rayman"
	"27"	"rappyraptor"
	"28"	"postal_dude"
	"29"	"phoenix"
	"30"	"painis_cupcake"
	"31"	"piss_cakehole"
	"32"	"new_ass_pancakes"
	"33"	"soupcock_porkpie"
	"34"	"new_duke_nukem"
	"35"	"pacman2"
	"36"	"new_announcer"
	"37"	"ghostbuster_slimer_v2"
	"38"	"gaben"
	"39"	"fazbear"
	"40"	"falcon"
	"41"	"donkey_kong"
	"42"	"Daftpunk"
	"43"	"Daftpunksilver"
	"44"	"crash3"
	"45"	"sonicexe"
	"46"	"Steve"
	"47"	"mlgsniper"
	"48"	"jamesbond"
	"49"	"claude"
	"50"	"serpentine"
	"51"	"triceratops"
	"52"	"new_weapons_soldier"
	"53"	"hatty"
	"54"	"onepunch"
	"55"	"bowserjr"
	"56"	"duckpyro"
	"57"	"buzzligyea"
	"58"	"drmario"
	"59"	"meeseeks"
	"60"	"ripperroo"
	"61"	"cuphead"
	"62"	"hatkid"
	"63"	"Pepsiman"
	"64"	"epicscout"
	"65"	"popukoduo"
	"66"	"pipimiduo"
	"67"	"dragonborn"
	"68"	"kirby"
	"69"	"ashley"
	"70"	"bomberman"
	"71"	"deathmachine"
}

Basically, more than the "Pepsiman" im having here, the rest of the bosses won't load.
Is there a way to increase the limit?

**Comment by Batfoxkid** [comment]
MAXSPECIALS in the main code defines the maximum number of bosses (minus one).
Changing that and recompiling changes the limit.

```sourcepawn
...
#define UPDATE_URL "http://50dkp.github.io/FF2-Official/update.txt"

#define MAXENTITIES 2048
#define MAXSPECIALS 64
#define MAXRANDOMS 16

#define SOUNDEXCEPT_MUSIC 0
...
```

**Comment by danylopez123** [comment]
Oohh.... i can see that... thanks for letting me know!

**Comment by naydef** [comment]
MAXSPECIALS is set to 64 with a reason(probably old reason, limited precache limits). Probably this is no longer an issue and can be raised to 128.

**Comment by naydef** [comment]
Adresses in #474
